src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/EditingTargetCodeExecution.java

```diff
@@ -158,6 +161,44 @@ public void executeRange(Range range)
       executeRange(range, null, false);
    }
    
+   public void sendSelectionToTerminal(boolean terminalExecuteWhenNotFocused)
+   {
+      if (terminalExecuteWhenNotFocused && !docDisplay_.isFocused())
+      {
+         // the current editor may be in another window; if one of our source
+         // windows was last focused, use that one instead
+         SourceWindowManager manager = 
+               RStudioGinjector.INSTANCE.getSourceWindowManager();
+         if (!StringUtil.isNullOrEmpty(manager.getLastFocusedSourceWindowId()))
+         {
+            RStudioGinjector.INSTANCE.getSatelliteManager().dispatchCommand(
+                  commands_.sendToTerminalWithoutFocus(), SourceSatellite.NAME_PREFIX + 
+                           manager.getLastFocusedSourceWindowId());
+            return;
+         }
+      }
+
+      // send selection from this editor to the terminal
+      Range selectionRange = docDisplay_.getSelectionRange();
+      boolean noSelection = selectionRange.isEmpty();
+      if (noSelection)
+      {
+         // current line
+         int row = docDisplay_.getSelectionStart().getRow();
+         selectionRange = Range.fromPoints(
+               Position.create(row, 0),
+               Position.create(row, docDisplay_.getLength(row)));
+
+         // if we failed to discover a range, bail
+         if (selectionRange == null)
```

Comment:
I don't think this would ever be `null`, would it? (`Range.fromPoints()` shouldn't ever produce a `null` value)